src/lib/layout/index.spec.ts

```diff
@@ -16,7 +15,62 @@ rootLogger.settings({
   },
 })
 
-const ctx = createTestContext()
+/**
+ * Helpers
+ */
+
+// In-memory file tree
+type VirtualFS = {
+  [path: string]: string | VirtualFS
+}
+
+const layoutContext = TestContext.create((opts: { tmpDir: () => string }) => {
+  return {
+    setup(vfs: VirtualFS) {
+      const tmpDir = opts.tmpDir()
+
+      writeToFS(tmpDir, vfs)
+    },
+    async scan() {
+      const tmpDir = opts.tmpDir()
```

Comment:
The thunk needs to be evaluated in the functions and not above otherwise it'll be of type `null`

Reply:
> Can it do so in beforeAll? beforeEach?

Just tried. So it works fairly intuitively but still super error-prone.

| dependant lifecycle | dependent lifecycle in which the dependency can be |   |
|---------------------|----------------------------------------------------|---|
| `beforeEach`        | `beforeEach`                                       |   |
| `beforeAll`         | `beforeEach` & `beforeAll`                         |   |